daemon/containerd/image_commit.go

```diff
@@ -117,7 +117,19 @@ func (i *ImageService) CommitImage(ctx context.Context, cc backend.CommitConfig)
 			return "", fmt.Errorf("failed to create new image: %w", err)
 		}
 	}
-	return image.ID(img.Target.Digest), nil
+	id := image.ID(img.Target.Digest)
+
+	c8dImg, err := i.NewImageManifest(ctx, img, commitManifestDesc)
+	if err != nil {
+		return id, err
+	}
+	if err := c8dImg.Unpack(ctx, container.Driver); err != nil {
+		if !cerrdefs.IsAlreadyExists(err) {
+			return id, fmt.Errorf("failed to unpack image: %w", err)
+		}
+	}
```

Comment:
We're committing a running container into an image which is a copy of the base image with an additional layer (archive of the container's RW layer).

We can't reuse the snapshot because the unpacked snapshot we have here is an RW layer (active snapshot) of the container (with the ID of the container), and for the image we need to have a non-RW (committed snapshot) with the ID matching the ChainID of the new image (BaseImageChainID | NewLayerDiffID).

I don't think we can avoid this, as we can't just `Commit` the RW layer snapshot into a new snapshot, because that would delete  it.
I don't think that's a problem though, as we don't really unpack all the image layers (because their snapshots already exists as the base image of the container had to be unpacked), just the last layer we just created from the container diff.

Reply:
> We can't reuse the snapshot because the unpacked snapshot we have here is an RW layer

Oh, good point yes, forgot about that for a minute.